app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -606,15 +606,29 @@ public void handleIntent(@NonNull final Intent intent) {
         if (intent.hasExtra(PLAYBACK_QUALITY)) {
             setPlaybackQuality(intent.getStringExtra(PLAYBACK_QUALITY));
         }
-
+//TODO reduce code duplication
         // Resolve append intents
-        if (intent.getBooleanExtra(APPEND_ONLY, false) && playQueue != null) {
+        if (intent.getBooleanExtra(APPEND_ONLY, false)
```

Comment:
I don't think `APPEND_ONLY` was created for this reason. The result of `intent.getBooleanExtra(APPEND_ONLY, false)` is always either `true` or `false`, so when `playQueue != null` one of the two if branches will surely be executed and `return;` will be reached. `APPEND_ONLY=false` doesn't mean "enqueue next", though I get the confusion (the player code is a mess :sob:). I'd do this instead:
- Rename `APPEND_ONLY` to `ENQUEUE` and use it for this `if`, where previously `APPEND_ONLY` was used
- Add `ENQUEUE_NEXT` and use it for the `if` below (the new one you added)

Reply:
9bab471ff07cde5cf9ca73b762221adfa27788ed does this.
I have not only renamed `APPEND_ONLY` but also corrected the String to be `enqueue`.
No String named `APPEND_ONLY` remains in the Player Class.